src/cli/hook-utils/get-check-update-data.spec.js

```diff
@@ -0,0 +1,389 @@
+    assert.deepEqual(jsonDepFiles, ['package.json']);
+
+    mockfs.restore();
+  });
+
+  // test if package.json can't be found
+  it(`returns an empty array of dependencies (just package.json) if it doesn't exist`, async () => {
+    const output = await importCheckUpdateDataMock();
+    // Mock Bolt JS file system
+    mockfs({
+      'test-project': {
+        'README.md': '1',
+        '.github': {
+          /** empty directory */
+        },
+        manifest: {
+          /** empty directory */
+        },
+        node_modules: {
+          /** empty directory */
+        },
+        triggers: {
+          /** empty directory */
+        },
+        '.eslintignore': '1',
+        '.eslintrc.json': '1',
+        '.gitignore': {
+          /** empty directory */
+        },
+        'app.js': '1',
+        LICENSE: '1',
+        'package-lock.json': '1',
+        'slack.json': '1',
+      },
+    });
+
+    const cwd = 'test-project';
+    // Should not error
+    const shouldNotThrow = async () => await output.getJSONFiles(`${cwd}`);
+    assert.doesNotThrow(shouldNotThrow);
+
+    // Get JSON dependency file (package.json) and make sure it records it
+    const { jsonDepFiles, inaccessibleFiles } = await output.getJSONFiles(
+      `${cwd}`
+    );
+    assert.isEmpty(jsonDepFiles);
+    assert.deepEqual(jsonDepFiles, []);
+
+    mockfs.restore();
+  });
+
+  // Test if package.json can be read
+  it('returns JSON file if package.json exists', async () => {
+    const { getJSON } = await importCheckUpdateDataMock();
+
+    // Mock Bolt JS file system
+    mockfs({
+      'test-project': {
+        'README.md': '1',
+        '.github': {
+          /** empty directory */
+        },
+        manifest: {
+          /** empty directory */
+        },
+        node_modules: {
+          /** empty directory */
+        },
+        triggers: {
+          /** empty directory */
+        },
+        '.eslintignore': '1',
+        '.eslintrc.json': '1',
+        '.gitignore': {
+          /** empty directory */
+        },
+        'app.js': '1',
+        LICENSE: '1',
+        'package.json': mockfs.load('./src/cli/hook-utils/test-json/test.json'),
+        'package-lock.json': '1',
+        'slack.json': '1',
+      },
+    });
+
+    const cwd = 'test-project';
+    // Should not error
+    const shouldNotThrow = async () => await getJSON(`${cwd}/package.json`);
+    assert.doesNotThrow(shouldNotThrow);
+
+    // Check returned package.json data and make sure it's valid
+    const jsonData = await getJSON(`${cwd}/package.json`);
+    assert.isNotEmpty(jsonData);
+    assert.containsAllKeys(jsonData, ['dependencies']);
+
+    mockfs.restore();
+  });
+
+  // Test for successful version map that needs upgrade
+  // TODO: getting error here, will need to add messaging to assert messages
+  it('returns a version map indicating it needs upgrades if it can access package.json and finds all dependencies', async () => {
```

Comment:
If there are any suggestions to simplify these last few tests, it is highly welcomed! They're running a bit longer than the others because they're using Sinon stubbing - I tried using `rewire` and wasn't able to mock function dependencies within the same file, but would love any suggestions to potentially either simplify that mocking or just simplify the tests in general! 
<img width="1138" alt="Screen Shot 2022-10-10 at 4 28 07 PM" src="https://user-images.githubusercontent.com/12901850/194949128-0c73e081-8cdf-430e-9bb0-3f08741a8690.png">

Reply:
I think the issue is your call to `sinon.stub` below happens _after_ the first call to `checkForSDKUpdates` (line 189) - so on line 189 we are actually interacting with npm over the network. That's my guess as to why that's slow!